src/dmd/traits.d

```diff
@@ -502,6 +502,11 @@ extern (C++) Expression semanticTraits(TraitsExp e, Scope* sc)
     }
     if (e.ident == Id.isDeprecated)
     {
+        if (global.params.vcomplex)
+        {
+            if (isTypeX(t => t.iscomplex() || t.isimaginary()))
+                return True();
+        }
         return isDsymX(t => t.isDeprecated());
```

Comment:
Why doesn't `return isDsymX(t => t.isDeprecated());` return true?

Reply:
Because `checkDeprecated` in `DSymbol` doesn't know about `checkComplexTransition` from `Type`. A `Type` also doesn't have the `isDeprecated` method, only `DSymbols` do. Most notably `AggregateDeclaration`.

Reply:
Ok, I looked closer at DMD's code and I don't see a straight-forward way to give `isDeprecated` semantics to `Type`, as it always seems to rely  on `Scope`.  Anyway, won't your proposed changes be removed when the feature is removed?  If so, please add one of your @@DEPRECATED@@ comments or something like that so we know it has an expiration date.

Reply:
Well for as long as we support complex numbers, this will need to be true. Unfortunately no expiration date is known for that. When support for complex numbers is finally removed from the compiler the global parameter and the `iscomplex` function will go, so the compiler will tell you about this block here. Imho a lot better than a deprecated comment.
Though adding a comment doesn't hurt and I can so later.

Reply:
> Though adding a comment doesn't hurt and I can so later.

No need.  Thanks for clarifying.